From 882b4044b23fc4b5533faf8592c84f6b3318e577 Mon Sep 17 00:00:00 2001
From: Kostas Papadakis <86578720+papadakis-k@users.noreply.github.com>
Date: Tue, 28 Nov 2023 03:31:49 +0200
Subject: [PATCH] additionally check for TERM variable when INSIDE_EMACS

This is done in order to differentiate between the dumb shell modes
shell and eshell, and the terminals term and vterm.
---
 src/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index 41f0e4c..bdf36ef 100644
--- a/src/main.c
+++ b/src/main.c
@@ -223,8 +223,8 @@ main(int argc, char **argv)
   
   run_unit_test(argc - optind, argv + optind, TEST_AFTER_OPTION_PARSING); /* argv points at the first non-option rlwrap argument */
 
-  /* if stdin is not a tty, or we're inside emacs, just execute <command>: */ 
-  if (!isatty(STDIN_FILENO) || getenv("INSIDE_EMACS"))
+  /* if stdin is not a tty, or we're inside emacs shell/eshell, just execute <command>: */
+  if (!isatty(STDIN_FILENO) || (getenv("INSIDE_EMACS") && strcmp(getenv("TERM"),"dumb") == 0))
     if (execvp(argv[optind], &argv[optind]) < 0)
       myerror(FATAL|USE_ERRNO, "Cannot execute %s", argv[optind]);
   
